ribbon/src/test/java/feign/ribbon/RibbonClientTest.java

```diff
@@ -98,6 +124,83 @@ public void ioExceptionRetry() throws IOException, InterruptedException {
     // assertEquals(target.lb().getLoadBalancerStats().getSingleServerStat())
   }
 
+  @Test
+  public void ioExceptionFailsAfterTooManyFailures() throws IOException, InterruptedException {
+    server1.enqueue(new MockResponse().setSocketPolicy(SocketPolicy.DISCONNECT_AT_START));
+    server1.enqueue(new MockResponse().setBody("success!"));
+
+    getConfigInstance().setProperty(serverListKey(), hostAndPort(server1.url("").url()));
+
+    TestInterface
+            api =
+            Feign.builder().client(RibbonClient.create()).retryer(Retryer.NEVER_RETRY)
+                    .target(TestInterface.class, "http://" + client());
+
+    try {
+      api.post();
+      fail("No exception thrown");
+    } catch (RetryableException ignored) {
+
+    }
+    assertEquals(1, server1.getRequestCount());
+    // TODO: verify ribbon stats match
```

Comment:
we should stop copy/pasting this TODO :) after however many years, I think we can give up on that.